doc/working_example.md

```diff
@@ -0,0 +1,825 @@
+```
+```{note}
+Genalyzer supports three window types that can be applied to the signal prior to computing FFT: Blackman-Harris, Hanning, and rectangular. The enumerations that map to these window types are shown [here](#genalyzer.advanced.advanced.Window).
+```
+```{important}
+Genalyzer doesn't support an overlap window between different snapshots that are averaged to generate the FFT. As a result, ``fft()`` expects the number of complex-valued samples in the input to be equal to the number of averages times the FFT order.
+```
+```{seealso}
+Genalyzer's ``fft()`` computes FFT for complex-valued data only. To compute FFT for real-valued data, use ``rfft()``. Additional details [here](#genalyzer.advanced.advanced.rfft).
+```
+<br/><br/>
+
+### Conduct Spectral Analysis
+Before conducting any spectral analysis, we configure a _manager_ to store _keys_ (more details follow in the subsections below). Only then, we run spectral analysis to calculate RF performance metrics. 
+#### Configure Genalyzer
+The following expanded code-snippet shows how Genalyzer is configured.
+<details>
+  <summary>Code Snippet</summary>
+
+``` Python
+import genalyzer.advanced as gn
+
+#
+# Signal 
+#
+npts = 30000 # number of points in the signal
+freq = 300000 # tone frequency
+phase = 0.0 # tone phase
+ampl_dbfs = -1.0 # amplitude of the tone in dBFS
+qnoise_dbfs = -63.0  # quantizer noise in dBFS
+fsr = 2.0 # full-scale range of I/Q components of the complex tone
+ampl = (fsr / 2) * 10 ** (ampl_dbfs / 20) # amplitude of the tone in linear scale
+qnoise = 10 ** (qnoise_dbfs / 20) # quantizer noise in linear scale
+qres = 12  # data resolution
+code_fmt = gn.CodeFormat.TWOS_COMPLEMENT # integer data format
+fs = 3e6 # sample-rate of the data
+
+#
+# FFT configuration
+#
+navg = 1 # number of FFT averages
+nfft = int(npts/navg) # FFT-order
+qres = 12  # data resolution
+code_fmt = gn.CodeFormat.TWOS_COMPLEMENT # integer data format
+window = gn.Window.NO_WINDOW # window function to apply
+
+#
+# Genarate signal for analysis
+#
+awfi = gn.cos(npts, fs, ampl, freq, phase)
+awfq = gn.sin(npts, fs, ampl, freq, phase)
+qwfi = gn.quantize(awfi, fsr, qres, qnoise, code_fmt)
+qwfq = gn.quantize(awfq, fsr, qres, qnoise, code_fmt)
+
+#
+# Compute FFT
+#
+fft_cplx = gn.fft(qwfi, qwfq, qres, navg, nfft, window, code_fmt)
+
+#
+# Fourier analysis configuration
+#
+object_key = "fa" # object-key
+signal_tone_key = 'A' # signal-key
+ssb_fund = 0 # number of single-side bins
+gn.fa_create(object_key)
+gn.fa_fsample(object_key, fs)
+gn.fa_max_tone(object_key, signal_tone_key, gn.FaCompTag.SIGNAL, ssb_fund)
+```
+</details>
+
+##### Create ``fourier_analysis`` Object
+The first step in configuring Genalyzer is to call the ``fa_create()`` function with an _object\_key_. This key is used for all configuring Genalyzer further, and for computing and retrieving the metrics.
```

Comment:
Aren't we on like step 2 at this point?

>This key is used for all configuring Genalyzer further

I think you meant configuration and not configuring

Can we use some other name than "object_key"? Is there something more relatable we can use?